packages/cli/src/commands/account/list.ts

```diff
@@ -8,29 +8,38 @@ export default class AccountList extends BaseCommand {
 
   static flags = {
     ...BaseCommand.flags,
+    local: flags.boolean({
+      allowNo: true,
+      description:
+        'If set, only show local and hardware wallet accounts. Use no-local to only show keystore addresses.',
+    }),
   }
 
   requireSynced = false
 
   async run() {
-    this.parse(AccountList)
+    const res = this.parse(AccountList)
 
+    // Retreive accounts from the connected Celo node.
+    const nodeAddresses = !res.flags.local ? await this.kit.web3.eth.getAccounts() : []
+
+    // Get addresses from the local wallet.
     const celoProvider: CeloProvider = this.kit.web3.currentProvider as any
-    const addresses = await this.kit.web3.eth.getAccounts()
-    const localAddresses = celoProvider.wallet
-      .getAccounts()
-      .map((value) => toChecksumAddress(value))
-    let localName = 'Local'
-    const res: ParserOutput<any, any> = this.parse()
+    const localAddresses =
+      res.flags.local ?? true
+        ? celoProvider.wallet.getAccounts().map((value) => toChecksumAddress(value))
+        : []
 
-    if (res.flags.useLedger) {
-      localName = 'Ledger'
+    // Display the addresses.
+    const localName = res.flags.useLedger ? 'Ledger' : 'Local'
+    if (res.flags.local === undefined) {
+      console.log('All Addresses: ', nodeAddresses.concat(localAddresses))
```

Comment:
The `this.kit.web3.eth.getAccounts()` retrieves the node + the local, because basically we are showing the user the addresses that have with all the info to sign a Tx (locally or via a node, but for the user it doesn't matter).
so, `nodeAddresses.concat(localAddresses)` will duplicate the local ones
That's why I was removing the locals from all the address to show the "keystore addresses" (or node addresses)